packages/kuma-gui/src/app/x/components/x-i18n/XI18n.vue

```diff
@@ -1,35 +1,100 @@
 <template>
-  <div
-    v-html="_t(props.t, params)"
-  />
   <template
-    v-for="(_slot, key) in slots"
-    :key="key"
+    v-if="props.path.length > 0"
   >
-    <XTeleportTemplate
-      :to="{ name: `x-i18n-${id}-${key}`}"
+    <!-- eslint-disable vue/no-v-html -->
+    <div
+      v-html="safeT(props.path, props.params)"
+    />
+    <!-- eslint-enable -->
+    <template
+      v-for="(_slot, slotName) in slots"
+      :key="slotName"
     >
       <slot
-        :name="key"
+        v-if="slotName === 'default'"
+        :name="slotName"
+        :t="t"
       />
-    </XTeleportTemplate>
+      <XTeleportTemplate
+        v-else
+        :to="{ name: `${id}-${slotName}`}"
+      >
+        <slot
+          :t="t"
+          :name="slotName"
+        />
+      </XTeleportTemplate>
+    </template>
+  </template>
+  <template
+    v-else
+  >
+    <slot
+      name="default"
+      :t="t"
+    />
   </template>
 </template>
 <script lang="ts" setup>
-import { useSlots, computed } from 'vue'
+// eslint-disable-next-line vue/prefer-import-from-vue
+import { escapeHtml } from '@vue/shared'
+import { useSlots } from 'vue'
+
+import { useI18n, uniqueId, useEnv } from '@/app/application'
+import createI18n from '@/app/application/services/i18n/I18n'
+
 
-import { useI18n, uniqueId } from '@/app/application'
-const { t: _t } = useI18n()
-const props = defineProps<{
-  t: string
-}>()
 const slots = useSlots()
 const id = uniqueId('x-i18n')
 
-const params = computed(() => {
-  return Object.keys(slots).reduce<Record<string, string>>((prev, key) => {
-    prev[key] = `<span data-x-teleport-id="x-i18n-${id}-${key}"></span>`
-    return prev
-  }, {})
+const icuEscapeHtml = (str: string) => str.replace(/</g, "'<'")
+  .replace(/%7B/g, '{')
+  .replace(/%7D/g, '}')
+
+const props = withDefaults(defineProps<{
+  strings?: Record<string, string> | ((escape: typeof icuEscapeHtml) => Record<string, string>)
+  prefix?: string
+  path?: string
+  params?: Record<string, string>
+}>(), {
+  strings: undefined,
+  prefix: '',
+  path: '',
+  params: () => ({}),
 })
+
+const i18n = typeof props.strings !== 'undefined' ? createI18n(typeof props.strings === 'function' ? props.strings(icuEscapeHtml) : props.strings, useEnv()) : useI18n()
+
+type ParametersExceptFirst<T> = T extends (_arg1: any, ...rest: infer R) => any ? R : never
+type ParametersExceptFirstTwo<T> = T extends (_arg1: any, _arg2: any, ...rest: infer R) => any ? R : never
+
+const t = (key: Parameters<typeof i18n['t']>[0], ...rest: ParametersExceptFirst<typeof i18n['t']>) => {
+  return i18n.t(`${key.startsWith('.') ? props.prefix : ''}${key}`, ...rest)
+}
+const safeT = (
+  key: Parameters<typeof i18n['t']>[0],
+  params: Parameters<typeof i18n['t']>[1] = {},
+  ...rest: ParametersExceptFirstTwo<typeof i18n['t']>
+) => {
+  // escape any param values, if the value isn't a string, empty it out we
```

Comment:
Just a nit, but I think we don't need to type each parameter individually via `Parameters<...>[...]` and `ParametersExcept...` if we just add a type to the whole function, like

```suggestion
const t: typeof i18n['t'] = (key, ...rest) => {
  return i18n.t(`${key.startsWith('.') ? props.prefix : ''}${key}`, ...rest)
}
const safeT: typeof i18n['t'] = (
  key,
  params = {},
  ...rest
) => {
  // escape any param values, if the value isn't a string, empty it out we
```

Reply:
Ok gotcha, lemme take another look at this brb

Reply:
Ok, I aliased the `typeof i18n['t']` to TFunction (the space there made it look weird to me), but you;re right this is simpler 👍 . Lemme know

(I also fixed up the `defineSlots` changes)